pkg/controller/knicluster/knicluster_controller.go

```diff
@@ -74,15 +74,18 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		}
 	}
 
-	kniClusterName, kniClusterNamespace := getKNICustomResourceNames()
+	kni, err := getKNINamespacedName()
+	if err != nil {
+		return err
+	}
 	err = c.Watch(&source.Kind{Type: &osconfigv1.ClusterVersion{}}, &handler.EnqueueRequestsFromMapFunc{
 		ToRequests: handler.ToRequestsFunc(
 			// always enqueued the same KNICluster object, since there should be only one
 			func(a handler.MapObject) []reconcile.Request {
 				return []reconcile.Request{
 					{NamespacedName: types.NamespacedName{
-						Name:      kniClusterName,
-						Namespace: kniClusterNamespace,
+						Name:      kni.Name,
```

Comment:
I think you're composing a `NamespacedName` from a `NamespacedName`. This could be collapsed into:

```
return []reconcile.Request{NamespacedName: kni}
```